src/panels/lovelace/editor/hui-dialog-edit-card.ts

```diff
@@ -1,229 +1,68 @@
-import "@polymer/paper-dialog-scrollable/paper-dialog-scrollable";
-import "@polymer/paper-dialog/paper-dialog";
+import { HomeAssistant } from "../../../types";
+import { LovelaceConfig } from "../types";
+import "./hui-edit-card";
 // This is not a duplicate import, one is for types, one is for element.
 // tslint:disable-next-line
-import { PaperDialogElement } from "@polymer/paper-dialog/paper-dialog";
-import { HomeAssistant } from "../../../types";
-import { getCardConfig, updateCardConfig } from "../common/data";
-import { fireEvent } from "../../../common/dom/fire_event";
-
-import "./hui-yaml-editor";
-import "./hui-yaml-card-preview";
+import { HuiEditCard } from "./hui-edit-card";
+import "./hui-migrate-config";
 // This is not a duplicate import, one is for types, one is for element.
 // tslint:disable-next-line
-import { HuiYAMLCardPreview } from "./hui-yaml-card-preview";
-import { LovelaceCardEditor, LovelaceConfig } from "../types";
-import { YamlChangedEvent, ConfigValue } from "./types";
-
-const CUSTOM_TYPE_PREFIX = "custom:";
+import { HuiMigrateConfig } from "./hui-migrate-config";
 
 export class HuiDialogEditCard extends LitElement {
-  protected hass?: HomeAssistant;
-  private _cardId?: string;
-  private _originalConfigYaml?: string;
-  private _configElement?: LovelaceCardEditor | null;
+  private _hass?: HomeAssistant;
+  private _cardConfig?: LovelaceConfig;
   private _reloadLovelace?: () => void;
-  private _editorToggle?: boolean;
-  private _configValue?: ConfigValue;
 
   static get properties(): PropertyDeclarations {
     return {
-      hass: {},
-      cardId: {
-        type: Number,
-      },
-      _dialogClosedCallback: {},
-      _configElement: {},
-      _editorToggle: {},
+      _hass: {},
+      _cardConfig: {},
     };
   }
 
-  public async showDialog({ hass, cardId, reloadLovelace }) {
-    this.hass = hass;
-    this._cardId = cardId;
-    this._reloadLovelace = reloadLovelace;
-    this._editorToggle = true;
-    this._configElement = undefined;
-    this._configValue = { format: "yaml", value: "" };
-    this._loadConfig().then(() => this._loadConfigElement());
-    // Wait till dialog is rendered.
-    await this.updateComplete;
-    this._dialog.open();
+  private get _editDialog(): HuiEditCard {
+    return this.shadowRoot!.querySelector("hui-edit-card")!;
   }
 
-  private get _dialog(): PaperDialogElement {
-    return this.shadowRoot!.querySelector("paper-dialog")!;
+  private get _migrateDialog(): HuiMigrateConfig {
+    return this.shadowRoot!.querySelector("hui-migrate-config")!;
   }
 
-  private get _previewEl(): HuiYAMLCardPreview {
-    return this.shadowRoot!.querySelector("hui-yaml-card-preview")!;
+  public async showDialog({ hass, cardConfig, reloadLovelace }): Promise<void> {
+    this._hass = hass;
+    this._cardConfig = cardConfig;
+    this._reloadLovelace = reloadLovelace;
+    await this.updateComplete;
+    this._cardConfig!.id
```

Comment:
`this.shadowRoot.children[0].showDialog()`

Reply:
Can typescript handle that?

Reply:
`(this.shadowRoot.children[0] as any).showDialog()`